### PR TITLE
Refresh assumed role credentials to avoid role chaining limitations

### PR DESCRIPTION
**Pull request by michael-dickinson-sainsburys**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by toniblyx — CHANGES_REQUESTED** [review]

File: include/assume_role [hunk 1]
```diff
@@ -64,6 +64,7 @@ assume_role(){
     export AWS_ACCESS_KEY_ID=$(cat $TEMP_STS_ASSUMED_FILE | jq -r '.Credentials.AccessKeyId')
     export AWS_SECRET_ACCESS_KEY=$(cat $TEMP_STS_ASSUMED_FILE | jq -r '.Credentials.SecretAccessKey')
     export AWS_SESSION_TOKEN=$(cat $TEMP_STS_ASSUMED_FILE | jq -r '.Credentials.SessionToken')
+    export AWS_SESSION_EXPIRATION=$(date -j -f "%Y-%m-%dT%H:%M:%SZ" "$(cat $TEMP_STS_ASSUMED_FILE | jq -r '.Credentials.Expiration')" "+%s")
```
[the remark this answers is not in the document]
Comment (michael-dickinson-sainsburys) [reply include/assume_role]: Updated the code to use the same mechanism for os detection as other functions

[the remark this answers is not in the document]
Comment (toniblyx) [reply include/assume_role]: thanks!

**Review by toniblyx — APPROVED** [review]

**Comment by toniblyx** [comment]
I had to revert this change because it fails when used with Assume role options `-A account -R role`, this is the error running on a Mac:

`./prowler: line 324: ((: < (( 1608050483 + (( 3600 / 10 )) )) : syntax error: operand expected (error token is "< (( 1608050483 + (( 3600 / 10 )) )) ")`

so this command fails on a Mac: `date -j -f %Y-%m-%dT%H:%M:%SZ 2020-12-15T17:46:14+00:00 +%s`

**Comment by toniblyx** [comment]
in Linux work fine.

**Comment by michael-dickinson-sainsburys** [comment]
I think it is the inclusion of so many parentheses, putting together a modified version that should run (and be significantly more readable)

**Comment by michael-dickinson-sainsburys** [comment]
#710 opened to try again

**Pull request merged**